### PR TITLE
Add capability to handle with node variables

### PR DESCRIPTION
**Pull request by bennibolm**
This PR adds the functionality to handle node-level data beside the solution itself.
This is required to visualize subcell limiting coefficients, as used in [#1476](https://github.com/trixi-framework/Trixi.jl/pull/1476) and [#1502](https://github.com/trixi-framework/Trixi.jl/pull/1502).

It comes together with the [PR to Trixi](https://github.com/trixi-framework/Trixi.jl/pull/1611), which saves the coefficients in the `.h5` files.

TODO:
- [x] Adapt test to "new" testing structure (compare value and not only check whether the code runs through). Check this after the Trixi PR is merged.

Question:
- [x] Useful to allow reinterpolation for subcell blending coefficients? Actually not, but it would be very annoying to always handle two files. Added a comment.

### CODE REVIEW

**Comment by bennibolm** [comment]
The current implementation supports both reinterpolation and no reinterpolation for the subcell limiting coefficient. That's the way it is handled for the solution itself.
Does it even make sense to allow reinterpolation for this coefficient? What do you think? @amrueda @andrewwinters5000 @sloede 

I already prepared an PR to `Trixi2Vtk_reference_files` to add the needed `.vtu` files for testing. I will open the PR when this discussion here has been decided.

**Comment by sloede** [comment]
> Does it even make sense to allow reinterpolation for this coefficient?

IMHO, no. It's like an AMR marker (should this cell be refined or not), which is not a continuous field that happens to be represented by a piecewise-constant approximation. Thus reinterpoaltion does not give a meaningful representation of reality but can actually be harmful.

Having said that, I am not sure how easy/annoying it is to always create two files, once once with reinterpolation (for the solution itself) and once without (for the node variables). It might thus be nice to have the ability for reinterpolation, just so you can have only one VTK file for a quick'n'dirty visualization.

What we really need/want though, is a ParaView plugin that can read in our HDF5 natively without us having to convert to VTK first. But that's for another day I guess :-)

**Comment by amrueda** [comment]
> It might thus be nice to have the ability for reinterpolation, just so you can have only one VTK file for a quick'n'dirty visualization.

I agree with @sloede

**Comment by bennibolm** [comment]
> > Does it even make sense to allow reinterpolation for this coefficient?
> 
> IMHO, no. It's like an AMR marker (should this cell be refined or not), which is not a continuous field that happens to be represented by a piecewise-constant approximation. Thus reinterpoaltion does not give a meaningful representation of reality but can actually be harmful.

I thought so too. But I implemented the reinterpolation of the coefficients before it was possible to not reinterpolate the solution and just kept the code after the update.
> 
> Having said that, I am not sure how easy/annoying it is to always create two files, once once with reinterpolation (for the solution itself) and once without (for the node variables). It might thus be nice to have the ability for reinterpolation, just so you can have only one VTK file for a quick'n'dirty visualization.
> 
I'm not sure that I understand your thoughts correctly. I don't think we need two different files here either way. It should be enough to just delete the implication of parameter `reinterpolation` for the subcell stuff.

> What we really need/want though, is a ParaView plugin that can read in our HDF5 natively without us having to convert to VTK first. But that's for another day I guess :-)

:relieved:

**Comment by amrueda** [comment]
> I'm not sure that I understand your thoughts correctly. I don't think we need two different files here either way. It should be enough to just delete the implication of parameter reinterpolation for the subcell stuff.

No, I don't think that's a good idea. In some situations, it might still be advantageous to reinterpolate the solution variables to get a nice and smooth representation of the (part of the field that is a pure) DG solution.

**Comment by bennibolm** [comment]
> Having said that, I am not sure how easy/annoying it is to always create two files, once once with reinterpolation (for the solution itself) and once without (for the node variables). It might thus be nice to have the ability for reinterpolation, just so you can have only one VTK file for a quick'n'dirty visualization.

>  > I'm not sure that I understand your thoughts correctly. I don't think we need two different files here either way. It should be enough to just delete the implication of parameter reinterpolation for the subcell stuff.
> No, I don't think that's a good idea. In some situations, it might still be advantageous to reinterpolate the solution variables to get a nice and smooth representation of the (part of the field that is a pure) DG solution.

After understanding the problem correctly:sweat_smile:, I will keep the current implementation for now and add a warning in that case, if that's okay for you.

**Comment by andrewwinters5000** [comment]
I also think that it not very "meaningful" to reinterpolate the node-wise `alpha` values because it could lead to misinterpretation (as already pointed out by the others). For the element-wise variant we can get away with this re-interpolation because the `alpha`s  are constant within each element, so it does not change the values.

> > Having said that, I am not sure how easy/annoying it is to always create two files, once once with reinterpolation (for the solution itself) and once without (for the node variables). It might thus be nice to have the ability for reinterpolation, just so you can have only one VTK file for a quick'n'dirty visualization.
> 
> I'm not sure that I understand your thoughts correctly. I don't think we need two different files here either way. It should be enough to just delete the implication of parameter `reinterpolation` for the subcell stuff.

I think that @sloede is correct. It would be annoying but you could create two separate files where the solution re-interpolates but the `alpha`s do not. It would be something like (note I have not tested this)
```julia
trixi2vtk(joinpath("out", "solution_*.h5"), output_directory="out")
trixi2vtk(joinpath("out", "solution_*_celldata.h5"), output_directory="out", reinterpolate=false)
```
The second call might need to use `?` instead of the `*` for the different files numbers.

**Comment by bennibolm** [comment]
So, to avoid the annoying part with two files, I just added a comment if `reinterpolation=true` is called with a subcell coefficient.

For the testing, I added two new tests. After running `elixir_euler_shockcapturing_subcell.jl` I check for `reinterpolation=true` and `false`. I would suggest, that I first request a review on the current changes here and then, after every is okay here, I open a PR to `Trixi2Vtk_reference_files` to add the files and (hopefully) fix both new tests.

File: test/test_2d.jl [hunk 1]
```diff
@@ -395,6 +395,43 @@ end
...
+
+      # save output file to `artifacts` to facilitate debugging of failing tests
+      testname = "2d-treemesh-shockcapturing-subcell-reinterp"
+      cp(out_file, joinpath(artifacts_dir, testname * "-" * outfilename), force=true)
+
+      # remote file path is actually a URL so it always has the same path structure
+      remote_filename = "2d/treemesh/dgsem_sedov_subcell_interp_10.vtu"
+      ref_file = get_test_reference_file("dgsem_sedov_subcell_interp_10.vtu", remote_filename)
+      compare_cell_data(out_file, ref_file)
+    end
```
Comment (bennibolm) [inline test/test_2d.jl:433]: Once I'm using `compare_point_data`, once `compare_cell_data`. the reason for that is:
```
julia> trixi_include(joinpath(examples_dir(), "tree_2d_dgsem", "elixir_euler_sedov_blast_wave_sc_subcell.jl"), maxiters=10)
julia> trixi2vtk("out/solution_*.h5", output_directory="out/", reinterpolate=false)
julia> compare_point_data("out/solution_000010.vtu", "out/solution_000010.vtu")
julia> compare_cell_data("out/solution_000010.vtu", "out/solution_000010.vtu")
BoundsError: attempt to access 0-element Vector{LightXML.XMLElement} at index [1]
Stacktrace:
 [1] getindex
   @ ./essentials.jl:13 [inlined]
 [2] get_data_section(vtk_file::VTKFile, section::String)
   @ ReadVTK ~/.julia/packages/ReadVTK/XpdR0/src/ReadVTK.jl:397
 [3] get_cell_data
   @ ~/.julia/packages/ReadVTK/XpdR0/src/ReadVTK.jl:417 [inlined]
 [4] compare_cell_data(out_filename::String, ref_filename::String; atol::Float64, rtol::Float64)
   @ Main ~/git/Trixi2Vtk.jl/test/test_trixi2vtk.jl:86
 [5] compare_cell_data(out_filename::String, ref_filename::String)
   @ Main ~/git/Trixi2Vtk.jl/test/test_trixi2vtk.jl:69
 [6] top-level scope
   @ REPL[23]:1
```
But with `reinterpolate=true` I get
```
julia> trixi_include(joinpath(examples_dir(), "tree_2d_dgsem", "elixir_euler_sedov_blast_wave_sc_subcell.jl"), maxiters=10)
julia> trixi2vtk("out/solution_*.h5", output_directory="out/", reinterpolate=true)
julia> compare_point_data("out/solution_000010.vtu", "out/solution_000010.vtu")
BoundsError: attempt to access 0-element Vector{LightXML.XMLElement} at index [1]
Stacktrace:
 [1] getindex
   @ ./essentials.jl:13 [inlined]
 [2] get_data_section(vtk_file::VTKFile, section::String)
   @ ReadVTK ~/.julia/packages/ReadVTK/XpdR0/src/ReadVTK.jl:397
 [3] get_point_data
   @ ~/.julia/packages/ReadVTK/XpdR0/src/ReadVTK.jl:444 [inlined]
 [4] compare_point_data(out_filename::String, ref_filename::String; atol::Float64, rtol::Float64)
   @ Main ~/git/Trixi2Vtk.jl/test/test_trixi2vtk.jl:142
 [5] compare_point_data(out_filename::String, ref_filename::String)
   @ Main ~/git/Trixi2Vtk.jl/test/test_trixi2vtk.jl:124
 [6] top-level scope
   @ REPL[29]:1
julia> compare_cell_data("out/solution_000010.vtu", "out/solution_000010.vtu")
```

Does that make sense? In other tests, it seems like `compare_point_data` works for reinterpolation and no reinterpolation.

File: src/convert.jl [hunk 2]
```diff
@@ -202,6 +202,26 @@ function trixi2vtk(filename::AbstractString...;
...
+              end
+              @timeit "interpolate data" interpolated_cell_data = interpolate_data(Val(format),
+                                                                    reshape(variable, size(variable)..., 1),
+                                                                    mesh, n_visnodes, verbose)
+            else
+              @timeit "interpolate data" interpolated_cell_data = reshape(variable,
+                                                                          n_visnodes^ndims_ * n_elements)
+            end
+            # Add the "interpolated" cell_data to celldata, not node_data
+            @timeit label vtk_nodedata[label] = interpolated_cell_data
```
Comment (sloede) [inline src/convert.jl:223]:
```suggestion
            # Add to node_data
            @timeit label vtk_nodedata[label] = interpolated_cell_data
```
?

File: src/convert.jl [hunk 3]
```diff
@@ -202,6 +202,26 @@ function trixi2vtk(filename::AbstractString...;
             verbose && println("| | Element variable: $label...")
             @timeit label vtk_celldata[label] = variable
           end
+
+          # Add node variables
+          for (label, variable) in node_variables
+            verbose && println("| | Node variable: $label...")
+            if reinterpolate
+              if index == 1
```
Comment (sloede) [inline src/convert.jl:210]: Why is this warning triggered by the `index`? Wouldn't it more sense to trigger it from the corresponding `label`?

File: test/test_2d.jl [hunk 4]
```diff
@@ -395,6 +395,43 @@ end
...
+      out_file = joinpath(outdir, outfilename)
+
+      # save output file to `artifacts` to facilitate debugging of failing tests
+      testname = "2d-treemesh-shockcapturing-subcell-reinterp"
+      cp(out_file, joinpath(artifacts_dir, testname * "-" * outfilename), force=true)
+
+      # remote file path is actually a URL so it always has the same path structure
+      remote_filename = "2d/treemesh/dgsem_sedov_subcell_interp_10.vtu"
+      ref_file = get_test_reference_file("dgsem_sedov_subcell_interp_10.vtu", remote_filename)
+      compare_cell_data(out_file, ref_file)
```
Comment (sloede) [inline test/test_2d.jl:432]: Is `compare_cell_data` the correct function call here? Just wondering, I actually don't know.

**Review by sloede — COMMENTED** [review]
LGTM with minor comments

File: src/convert.jl (hunk 2 above)
Comment (bennibolm) [reply src/convert.jl:223]: Okay, that doesn't make any sense. Thanks

File: src/convert.jl (hunk 3 above)
Comment (bennibolm) [reply src/convert.jl:210]: I agree with you to trigger the warning just for the label `:limiting_coefficient`. Unfortunately, this will be an issue when using it for MCL where we have labels like `Symbol("alpha_", variables[v])` for all variables v, `:alpha_pressure` and `:alpha_entropy`. Is it acceptable to add an if clause like `if contains("limiting", label) || contains("alpha", label)`?

Moreover, the if clause `if index == 1` is used to print the warning only once per call of `trixi2vtk`. Otherwise, for every chosen file, we get a warning. I could move this warning outside the for loop over all files to just print it once.

File: test/test_2d.jl (hunk 4 above)
Comment (bennibolm) [reply test/test_2d.jl:432]: I also was wondering about that. I wrote all my thoughts to this "discussion" https://github.com/trixi-framework/Trixi2Vtk.jl/pull/67#discussion_r1576160118. Did you read that?

File: src/convert.jl (hunk 2 above)
Comment (bennibolm) [reply src/convert.jl]: Done in [526b7ce](https://github.com/trixi-framework/Trixi2Vtk.jl/pull/67/commits/526b7cec93696b3416d83e173da4b27a0e8a91b1)

**Comment by bennibolm** [comment]
@sloede I adapted the hash number `TEST_REFERENCE_COMMIT` to the current commit in Trixi2Vtk_reference_files. So, the tests are actually testing now. But two issues now:
1) Some tests fail due to differences in the files. This is probably because I used julia 1.10 for creating the files. Is there a useful way to just take the file the CI created (and if so, where)? In the past, I had so many failing tests because of different machines/versions/..., such that I don't feel confident with just creating the reference files locally with julia 1.9.

2) In the test with julia version 1.7 the following error occurs:
`SystemError: opening file "/home/runner/.julia/packages/Trixi/enFLf/examples/tree_2d_dgsem/elixir_euler_sedov_blast_wave_sc_subcell.jl": No such file or directory`. Does it use an old version of Trixi without any subcell limiting elixirs? I haven't found a possible reason for that yet.
Also, for julia 1.7 only the 2d tests are running. Probably, when finding the reason for that, also explains why it gives the error above.

File: src/convert.jl (hunk 3 above)
Comment (sloede) [reply src/convert.jl:210]: Since Trixi2Vtk.jl is not going to win any code beauty contest, I'd be OK with having this special case for alpha and the limiting coefficient. Although I'd prefer to use ```julia
label == "limiting_coefficient" || startswith(label, "alpha_")
```
instead.

Regarding repeated warnings: just use a sentinel variable (`has_warned_about_interpolation = false`) that you set to `true` after the first warning and then don't warn anymore 🤷‍♂️

File: test/test_2d.jl (hunk 4 above)
Comment (sloede) [reply test/test_2d.jl:432]: Right. Interpolation is to piecewise constant VTK cells, while original data is high-order Lagrangian VTK cells.

File: src/convert.jl (hunk 3 above)
Comment (bennibolm) [reply src/convert.jl]: Done in [aeb6802](https://github.com/trixi-framework/Trixi2Vtk.jl/pull/67/commits/aeb6802cd7a8c58db19b8fc3f85afa6ec686ca9e)

**Comment by bennibolm** [comment]
> Re 1: Here you should be able to get the artifacts files for older runs: https://github.com/trixi-framework/Trixi2Vtk.jl/actions/runs/9328145281. I am not 100% sure how to get them if the initial run fails - maybe try to make the test pass artificially, download the reference files, and then update them?

Okay thanks. It seems like the artifacts are saved anywhere. So, I can directly download the files from https://github.com/trixi-framework/Trixi2Vtk.jl/actions/runs/9614040983. 

> I know this is a finicky business, but nobody has yet put in the time for a proper testing setup for Trixi2Vtk.jl... if you're up for it, we could sit together and try to come up with some strategies on how to mitigate this 😊

I'm not sure if I want to spend time on that :no_mouth: 

> 
> Re 2: Yes, Julia v1.7 heavily downgrades Trixi.jl. We switched to a minimum of Julia v1.8 over a year ago (https://github.com/trixi-framework/Trixi.jl/blame/06407f44041a04352800b7add4b20a9ac235f4ae/Project.toml), so your tests probably require Juila v1.8 or later. I'd just put the tests in an if condition for the Julia version to check if that fixes it.

Okay thanks. Will do it.

**Comment by sloede** [comment]
> > I know this is a finicky business, but nobody has yet put in the time for a proper testing setup for Trixi2Vtk.jl... if you're up for it, we could sit together and try to come up with some strategies on how to mitigate this 😊
> 
> I'm not sure if I want to spend time on that 😶

I totally understand. Just keep in mind that all this infrastructure around Trixi.jl, including pre-/postprocessing was built by someone who probably should have been doing other things. Not saying it needs to be you, but things like Trixi2Vtk.jl don't build themselves (yet 😬).

**Comment by bennibolm** [comment]
It worked with downloading the reference file from the artifact page. Unfortunately, the test fails on macOS.
I downloaded that vtu file from the last run of macOS as well, and sadly it seems like they differ significantly and the test only run through with very high tolerances.
```
julia> # downloaded file for macOS
julia> mac_file = "/home/bbolm/Downloads/test-output-files-1.9-macOS-latest-x64(1)/2d-treemesh-shockcapturing-subcell-reinterp-solution_000010.vtu"

julia> # current reference file (downloaded from ubuntu)
julia> ubuntu_file = "/home/bbolm/Downloads/test-output-files-1.9-ubuntu-latest-x64(1)/2d-treemesh-shockcapturing-subcell-reinterp-solution_000010.vtu"

julia> compare_cell_data(out_file, ref_file, atol=1.0e-12, rtol=1.0e-8)
ERROR: There was an error during testing

julia> compare_cell_data(out_file, ref_file, atol=1.0e-12, rtol=1.0e-3)
ERROR: There was an error during testing

julia> compare_cell_data(out_file, ref_file, atol=1.0e-12, rtol=1.0e-2)

julia> compare_cell_data(out_file, ref_file, atol=1.0e-2, rtol=1.0e-3)
ERROR: There was an error during testing

julia> compare_cell_data(out_file, ref_file, atol=1.0e-1, rtol=1.0e-3)
```
So, the abs tolerance has to be about `1.0e-1` or the rel tol `1e-3`. Can this be correct somehow?
At this point I'd rather remove the macOS test than use such tolerances...

And for the test without reinterpolation almost the same. There the tests work with `atol=1.0e-2` or `rtol=1.0e-2`.
```
julia> mac_file = "/home/bbolm/Downloads/test-output-files-1.9-macOS-latest-x64(1)/2d-treemesh-shockcapturing-subcell-no-reinterp-solution_000010.vtu"

julia> ubuntu_file = "/home/bbolm/Downloads/test-output-files-1.9-ubuntu-latest-x64(1)/2d-treemesh-shockcapturing-subcell-no-reinterp-solution_000010.vtu"
```

I don't really know what to do with that :confused: @sloede What so you think?

**Comment by sloede** [comment]
Do the differences in the solution between macOS and Ubuntu already exist before converting the files to VTK? That is, is the 1e-4 difference already present in the Trixi.jl solution (files)? And did you experience something similar for the Trixi.jl tests on macOS?

**Comment by bennibolm** [comment]
> Do the differences in the solution between macOS and Ubuntu already exist before converting the files to VTK? That is, is the 1e-4 difference already present in the Trixi.jl solution (files)? And did you experience something similar for the Trixi.jl tests on macOS?

That's a good question. No, within CI runs, I never really experienced differences between linux and macos in Trixi.jl before. What I experienced was, that the subcell limiting itself is very sensitive to small changed (per construction). So, small differences can grow bigger and bigger over time.
I just looked into the differences, and they occur for all variables (rho, v1, ...) and not only for the limiting coefficient.

Note: Since even the tests without any reinterpolation are failing for macOS, where all data is only copied and reshaped within Trixi2Vtk, it is not only a problem here but maybe in the compression in between.

I'm no expert on the Vtk files, but for me, it looks like the vtk file saves the data with type `UInt8`. [routine here](https://github.com/bennibolm/Trixi2Vtk.jl/blob/7dec585a60857b55b88023f54560c2ea2eb3b5af/test/test_trixi2vtk.jl#L124-L169)
```
julia> vtk = VTKFile(out_filename).appended_data
21262-element Vector{UInt8}
[...]
```
Then, in `get_point_data` it is converted back to `Float64`. 
```
julia> out_point_data = get_point_data(vtk)
julia> out_data = get_data(out_point_data[variable_name])
julia> out_data
1024-element reinterpret(Float64, ::Vector{UInt8}):
[...]
```
Is it possible that subcell limiting produces slightly different solutions on the different machines, which grow bigger due to the compression in the vtu file? Although, the largest difference for rho is about `3.0e-3`, which is quite large :thinking:

**Comment by sloede** [comment]
> Is it possible that subcell limiting produces slightly different solutions on the different machines, which grow bigger due to the compression in the vtu file? Although, the largest difference for rho is about `3.0e-3`, which is quite large

The compression is lossless, so I don't really think that this is the reason.



> Note: Since even the tests without any reinterpolation are failing for macOS, where all data is only copied and reshaped within Trixi2Vtk, it is not only a problem here but maybe in the compression in between.

But this indicates that the difference is already there in the Trixi.jl solution, i.e., before Trixi2Vtk touches it, doesn't it?

**Comment by bennibolm** [comment]
> > Note: Since even the tests without any reinterpolation are failing for macOS, where all data is only copied and reshaped within Trixi2Vtk, it is not only a problem here but maybe in the compression in between.
> 
> But this indicates that the difference is already there in the Trixi.jl solution, i.e., before Trixi2Vtk touches it, doesn't it?

Yes, that's true. I just took a look into the ci file in Trixi. Do I get this correctly that we don't run the "normal" tests on macos and windows? Only the threaded and mpi tests.
If so, that wasn't clear to me. Then, of course, it is highly likely that already in Trixi the results are quite different on macOS.
(When looking at the new PR including tests with julia 1.11 (with errors of about 1e-4) and remembering that we already had to adapt the tests for the move to 1.10, it is even more likely. Of course, both of it for linux)

In the past, I always had this issue, mostly with the blast elixirs. Not really sure why that is. Do you think I sould spend some time on investigating this? I mean, the method itself is very sensitive to small changes by construction.
A short-term fix for this current PR here, could be to accept the differences and exclude the test from ci for macOS. What do you think? @sloede

**Comment by sloede** [comment]
> Do I get this correctly that we don't run the "normal" tests on macos and windows? Only the threaded and mpi tests.

Yes. Since fewer macOS runners are available to us on the free plan, we only run some basic tests on macOS and assume that everything else "just works" as well. Originally, we tested everything everywhere, but this just wasn't feasible anymore due to long CI runtimes.

> Do you think I sould spend some time on investigating this? I mean, the method itself is very sensitive to small changes by construction.

This is something you need to discuss with your advisor, if it is worth to investigate right now (maybe bring it up on Friday?). IMHO it is at least somewhat unfortunate that floating point truncation errors seem to have such a significant impact already. It indicates to me that either there's something wrong with the method itself (unlikely), the implementation (less unlikely, but still), or that the setups are not robust enough (my current favorite). By robust I mean that maybe you are not running a stable simulation but rather simulations that are just not crashing. I wonder if choosing slightly different parameters would allow you to alleviate the issue (thus confirming my suspicion) or whether it is really by design.

> A short-term fix for this current PR here, could be to accept the differences and exclude the test from ci for macOS.

Yes. But please leave a short comment explaining the reason for this and x-ref this PR (https://github.com/trixi-framework/Trixi2Vtk.jl/pull/67)

**Comment by bennibolm** [comment]
> IMHO it is at least somewhat unfortunate that floating point truncation errors seem to have such a significant impact already. It indicates to me that either there's something wrong with the method itself (unlikely), the implementation (less unlikely, but still), or that the setups are not robust enough (my current favorite). By robust I mean that maybe you are not running a stable simulation but rather simulations that are just not crashing. I wonder if choosing slightly different parameters would allow you to alleviate the issue (thus confirming my suspicion) or whether it is really by design.

Yes, I agree. I took a quick look into it and made a list of failing tests. I will add this here but will present this also on friday:
Subcell Limiting in Trixi:

    • Failing tests for julia 1.11
        ◦ elixir_euler_blast_wave_sc_subcell_nonperiodic.jl 
          initial_condition_blast_wave
          local limiting of rho and entropy_math
          initial refinement level = 4 (in test)
          Dirichlet Boundary Conditions
        ◦ elixir_euler_sedov_blast_wave_sc_subcell.jl
          initial_condition_sedov_blast_wave
          local limiting of rho and entropy_guermond_etal
          initial_refinement_level=4
          periodic
        ◦ elixir_eulermulti_shock_bubble_shockcapturing_subcell_minmax.jl
          initial_condition_shock_bubble
          local limiting of rhos
          initial refinement level = 3
          periodic
    • Working tests:
        ◦ elixir_euler_shockcapturing_subcell.jl (Positivity of rho, initial_refinement_level = 5)
        ◦ elixir_euler_kelvin_helmholtz_instability_sc_subcell.jl (Positivity of rho and p, in_refine_level=5)
        ◦ elixir_eulermulti_shock_bubble_shockcapturing_subcell_positivity (Positivity of rhos, in_ref_level=3)
        ◦ elixir_mhd_shockcapturing_subcell (Positivity of rho and p, initial_refinement_level=4)
        ◦ elixir_euler_sedov_blast_wave_sc_subcell (StructuredMesh, Positivity of Rho and p, 16x16)
        ◦ elixir_euler_sedov_blast_wave_sc_subcell (StructuredMesh, Local limiting of rho and entropy_guermond, 16x16)
    • Used setup in tests of Trixi2Vtk
        ◦ elixir_euler_sedov_blast_wave_sc_subcell.jl (see above, local limiting, within failing tests for switch to julia 1.10 and 1.11)
    • Switch to julia 1.10
        ◦ elixir_euler_sedov_blast_wave_sc_subcell.jl
        ◦ elixir_eulermulti_shock_bubble_shockcapturing_subcell_minmax.jl

After that (but of course this must be investigated further) I would assume your are right with:

> [...] or that the setups are not robust enough (my current favorite). By robust I mean that maybe you are not running a stable simulation but rather simulations that are just not crashing. I wonder if choosing slightly different parameters would allow you to alleviate the issue (thus confirming my suspicion) [...]

**Review by sloede — DISMISSED** [review]
LGTM!

**Comment by sloede** [comment]
@bennibolm do you know why the tests do not seem to run?

**Comment by bennibolm** [comment]
> @bennibolm do you know why the tests do not seem to run?

@sloede I think, they are running, but are somehow showed very strange. When I press on the red X at the last commit, the CI test ran and there is an old version (failing) and a new version of the coverall tests.

I will try to make them run and ping you again.

**Comment by bennibolm** [comment]
@sloede I adapted the test setup for macOS. Is it possible that one needs to adapt the tests in GitHub that must pass in a PR? That could be the reason why they are not shown correctly but actually run properly.

**Review by sloede — APPROVED** [review]

**Comment by sloede** [comment]
I don't understand why the tests don't show up here, but as you pointed out correctly, they seem to have passed for the latest commit. Thus I'll merge this now.

**Comment by bennibolm** [comment]
Thanks
